### PR TITLE
Adds is_followed to show

### PR DESCRIPTION
**Pull request by ashfurrow**
Supersedes #1485. I modified the Gravity endpoint [on Matt's suggestion](https://github.com/artsy/metaphysics/pull/1485#discussion_r253689096) in [this PR](https://github.com/artsy/gravity/pull/12144). Fixes LD-184.

This is a WIP until the Gravity deploy is confirmed. I'm actually not sure how the process works for blocking PRs with dependent Gravity changes, but my gut says this should be fine to merge since it's only an addition.

### CODE REVIEW

**Comment by ashfurrow** [comment]
A staging deploy is on the way now, and then I can test locally.

**Comment by ashfurrow** [comment]
Okay so that's deployed but not working. I think this isn't working because of this line:

https://github.com/artsy/metaphysics/blob/dc8de9d4089f5c8e8c34cae9634b68759740a1a8/src/lib/loaders/loaders_with_authentication/tracked_entity.ts#L24

This is using `id` but it really means slug, and the [Gravity PR](
https://github.com/artsy/gravity/pull/12144) explicitly only allows IDs. When I change `${entityKeyPath}.id` to `${entityKeyPath}._id`, it works, so I'm going to look at changing `trackedEntityLoaderFactory` to accommodate strict IDs.

**Comment by ashfurrow** [comment]
Okay, this is working locally now 👍 I've updated the tracked entity tests to use async/await but I put that change in its own commit; you can review the non-refactoring changes here: https://github.com/artsy/metaphysics/pull/1497/files/3ef11506895e4a50d369b14fa1d161e283085999

File: src/lib/loaders/loaders_with_authentication/tracked_entity.ts [hunk 1]
```diff
@@ -9,19 +9,23 @@ import { map, find, extend } from "lodash"
...
  * @param {string} [entityKeyPath] an optional path to a nested entity
+ * @param {string} [entityIDKeyPath] an optional path to the IDs used to compare entities (defaults to `id`, but `_id` is sometimes useful).
  */
 const trackedEntityLoaderFactory = (
   dataLoader: (id: any) => Promise<any>,
   paramKey: string,
   trackingKey: string,
-  entityKeyPath?: string
+  entityKeyPath?: string,
+  entityIDKeyPath: string = "id"
```
Comment (ashfurrow) [inline src/lib/loaders/loaders_with_authentication/tracked_entity.ts]: This is what I had to do to get `_id` working – it seemed like the best approach but I'm open to suggestions.

Comment (mzikherman) [reply src/lib/loaders/loaders_with_authentication/tracked_entity.ts]: Oo, interesting bug, and yes adding an 'id path' with that default seems absolutely perfect to me!

File: src/lib/loaders/loaders_with_authentication/gravity.ts [hunk 2]
```diff
@@ -45,6 +45,13 @@ export default (accessToken, userID, opts) => {
     followShowLoader: gravityLoader("follow_shows", {}, { method: "POST" }),
     unfollowShowLoader: gravityLoader("follow_shows", {}, { method: "DELETE" }),
     followedShowsLoader: gravityLoader("follow_shows", {}, { headers: true }),
+    followedShowLoader: trackedEntityLoaderFactory(
+      gravityLoader("follow_shows"),
+      "show_ids",
```
Comment (mzikherman) [inline src/lib/loaders/loaders_with_authentication/gravity.ts]: Seeing this list of params (which can _definitely_ be confusing as to significance and ordering), makes me wish the factory took an options object.

``` typescript
type TrackedLoaderFactoryOptions {
  paramKey: string
  trackingKey: string
  entityKeyPath?: string
  entityIDKeyPath?: string
}
``` and so forth... but that's an improvement for another time!

**Comment by mzikherman** [comment]
Awesome! 😎

File: src/lib/loaders/loaders_with_authentication/gravity.ts (hunk 2 above)
Comment (ashfurrow) [reply src/lib/loaders/loaders_with_authentication/gravity.ts]: Yeah! It was getting a bit much with positional params. I think that'd be a good future improvement 👍

Comment (alloy) [reply src/lib/loaders/loaders_with_authentication/gravity.ts]: Agreed

**Pull request merged**